File: programming/EE1083/jee/2016/python_comp/codes/ee16b1022.py
import numpy as np
import matplotlib.pyplot as plt

#Point of intersection
A = np.array([[1,-1],[2,1]])
B = np.transpose(np.array([1,3]))
O = np.dot(np.linalg.inv(A),B)

#Finding radius
P = np.transpose(np.array([1,-1]))
r = np.linalg.norm(O-P)

#Intersection plot
x = np.linspace(-3,5,10000)
y = x - 1
z = 3 - 2*x
plt.axis('equal')
plt.plot(x,y,label = '$y - x + 1 = 0$')
plt.plot(x,z,label = '$y + 2x = 3$')
plt.grid()

theta = np.linspace(0,2*np.pi,100)
x = r*np.cos(theta) + 4.0/3
y = r*np.sin(theta) + 1.0/3

r = 3 - 4*x
s = -(3 + x)/4.0
t = 3*x - 4
u = (x - 4)/3.0

plt.plot(x,y)
plt.plot(x,r,label = '$y + 3x = 3$')
plt.plot(x,t,label = '$y - 3x + 4 = 0$')
plt.plot(x,u,label = '$3y - x + 4 = 0$')
plt.plot(x,s,label = '$4y + x + 3 = 0$')

plt.axis([-3,5,-3,3],'equal')
plt.legend()
plt.savefig('../figs/ee16b1022.eps')
plt.show()

File: programming/EE1083/jee/2016/octave_comp/codes/ee16b1022.m
clear;
close;
clc;

%Finding point of intersection
A = [1 -1; 2 1];
b = [1 3]';
O = inv(A)*b

%Finding the radius
P = [-1 1]';
r = norm(O-P)

%Intersection plot
x=linspace(-3,5,100);
y=x-1;
z=3-2*x;
plot(x,y,'g','LineWidth',4,x,z,'r','LineWidth',4)
axis("equal")
grid
hold
%clear;
%close;
%clc;
y=sqrt(53/9 - (x-(4/3)).^2)+1/3;
z=-(sqrt(53/9 - (x-(4/3)).^2))+1/3;
r=3-4*x;
s=-(3+x)./4;
t=3*x-4;
u=(x-4)./3;
plot(x,y,'LineWidth',4,x,z,'LineWidth',4,x,r,'LineWidth',4,x,s,'LineWidth',4,x,t,'LineWidth',4,x,u,'LineWidth',4)
axis([-3 5 -3 3])
legend('x-y=1','2x-y=3','circle','circle','4x+y=3','x+4y=3','3x-y=4','x-3y=4','location','northwest')
print ('ee16b1022.eps', '-color')
